### ISSUE TITLE
Support for multiple Homepage layouts and widgets configurations for the same installation

### ISSUE THREAD

**Issue by adaybujeda** [OSC/ondemand#2117]
At Harvard IQSS, we have a requirement to show different homepage layouts and widgets to different users. As well as the ability to switch between these layouts dynamically by the user.

**Proposal:**
PR: https://github.com/OSC/ondemand/pull/2118

Enhance the current configuration object to support profiles. These new profiles will work as overrides to the default root configuration. For every configuration item, first, there will be a lookup into the current profile configuration, then into the root configuration.

The new profile configuration will be part of the standard configuration files currently read by the system. We will add a new configuration key, `profiles`, to add these configurations under, eg for new profile `team1`:

`/etc/ood/config/ondemand.d/team1.yml`
```
profiles:
  team1:
    pinned_apps:
      - sys/RStudio
    dashboard_layout:
      rows:
        - columns:
            - width: 12
              widgets:
                - "pinned_apps"
```

To switch profiles, a new controller will be added. Current user profile will be stored into a new file in the data directory:` Configuration.dataroot`

Links can be configured in the OnDemand interface to dynamically select a profile and update the current configuration in use.

**Technical Notes:**
We currently have a prototype to demonstrate this idea.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1202561901230632) by [Unito](https://www.unito.io)

**Comment by adaybujeda** [comment]
Demo of our prototype:

![configuration](https://user-images.githubusercontent.com/17336210/177590693-3ff3b9ce-4f3d-42a0-9e99-1ea8c2f8311b.gif)

**Comment by johrstrom** [comment]
Thanks for the demo looks like lots of changes! 😆 I don't see anything completely crazy. Config options always give me pause - they have to be right because we basically can't change them after we publish & document.

I think that's the bit that'll get bogged down on. what's the best way to allow for configurations that's likely to stick (i.e., extendable but will not need to change significantly).

**Comment by adaybujeda** [comment]
The implementation is simpler that you might expect. I am simply adding an extra dimension to the existing configuration.
Created a PR with our current work so that you can take a look at the implementation:
https://github.com/OSC/ondemand/pull/2118

**Comment by johrstrom** [comment]
Internals can change all we like, but any config options/scheme we provide to sites has to be 100% solid or at least as good as it can be.

```
key_1: root_value_1
key_2: root_value_2
profiles:
  profile1:
    key_1: profile1_value_1
  profile2:
    key_1: profile2_value_1
```

Specifically this structure is what I mean. Now I'm not saying what you have is wrong or bad. I just think that'll be my sticking point. Making sure this is as right as it can be.

**Comment by johrstrom** [comment]
I believe this was fixed in #2118. Just open more tickets when you see fit.
